File: leetcode/2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.js
/**
 * @param {string[]} words
 * @param {string} target
 * @param {number} startIndex
 * @return {number}
 */
var closestTarget = function(words, target, startIndex) {
    const n = words.length;
    let minDist = n;
    
    for (let i = 0; i < n; i++) {
        if (words[i] === target) {
            const direct = Math.abs(i - startIndex);
            const wrap = n - direct;
            minDist = Math.min(minDist, Math.min(direct, wrap));
        }
    }
    
    return minDist === n ? -1 : minDist;
};

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.java
class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        int arrayLength = words.length;
        int minDistance = arrayLength;

        for (int currentIndex = 0; currentIndex < arrayLength; ++currentIndex) {
            if (words[currentIndex].equals(target)) {
                int directDistance = Math.abs(currentIndex - startIndex);
                int wrapAroundDistance = arrayLength - directDistance;
                int shortestPath = Math.min(directDistance, wrapAroundDistance);
                minDistance = Math.min(minDistance, shortestPath);
            }
        }

        return minDistance == arrayLength ? -1 : minDistance;
    }
}